src/components/CourseList/CourseList.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { useAppSelector } from '../../app/hooks';
+import { Courses } from '../../types/Courses';
+import { CourseCard } from '../CourseCard';
+import { Pagination } from '../Pagination';
+import './CourseList.scss';
+
+export const CourseList: React.FC = () => {
+  const { courses } = useAppSelector(state => state.courses);
+
+  const [searchParams] = useSearchParams();
+
+  const [leftLimit, setLeftLimit] = useState(0);
+  const [rightLimit, setRightLimit] = useState(courses.length);
+
+  const currentPage = searchParams.get('page') || '1';
+  const perPage = 10;
+
+  useEffect(() => {
+    const maxRightLimit = perPage * +currentPage;
+
+    if (maxRightLimit > courses.length) {
+      setRightLimit(courses.length);
+    } else {
+      setRightLimit(maxRightLimit);
+    }
```

Comment:
Maybe it would be a bit simplier to rewrite like that:
setRightLimit(maxRightLimit > courses.length ? courses.length : maxRightLimit);